### ISSUE TITLE
test failure

### ISSUE THREAD

**Issue by tlhackque** [perltidy/perltidy#142]
I still keep a 5.8.8 Perl around.  Not sure if you'd want to invest effort in this antique, but FYI:

Installing latest perltidy fails with:
```
t/snippets29.t ............ 1/19 Error output received for test 'dia.dia3'
An error flag '1' was returned
t/snippets29.t ............ 7/19
#   Failed test at t/snippets29.t line 717.
---------------------
<<STDERR>>
][ }->{ ]->{ }->[
 ^
Error parsing --interbracket-arrow-style: unexpected ']'
---------------------

#   Failed test at t/snippets29.t line 723.
t/snippets29.t ............ 15/19 # Looks like you planned 19 tests but ran 20.
# Looks like you failed 2 tests of 20 run.
t/snippets29.t ............ Dubious, test returned 2 (wstat 512, 0x200)
```
Your prereqs list 5.008, and perltidy seems happy running on 5.8.8 if I install it anyway.  So it looks like simply a test issue.

I'm happy ignoring the failure, though you might skip this test under Perls that can't run it, or even update your prereqs.  I can always use perlbrew to tidy under a newer perl...

**Comment by shancock9** [comment]
Thanks for letting me know.I would like to support versions back to 5.08 as long as possible.

I used perlbrew with version 5.8.8 and found the problem.  I need to add some scope-limiting
parens in a new regex which parses the parameter --interbracket-arrow-style. The problem
does not occur in Perl versions 5.10 and later. Version 20240511 will run on versions earlier
than 5.10 as long as this new -ias flag is not used.

I will upload a tagged version with the fix to Github later today. I'm not sure if it is worth also
uploading a fixed version to CPAN, but I can if anyone running an old version of perl needs one.

Thanks again!

**Comment by shancock9** [comment]
This problem is fixed in tagged version  20240511.01 at

https://github.com/perltidy/perltidy/tags

**Comment by shancock9** [comment]
CPAN version 20240903 of perltidy successfully installs on Perl 5.8.1. Thanks again for reporting this issue.
